class.jetpack-jitm.php

```diff
@@ -43,27 +44,42 @@ private function __construct() {
 	 */
 	function prepare_jitms( $screen ) {
 		global $pagenow;
-		$post_type = $screen->post_type;
+		// Only show auto update JITM if auto updates are allowed in this installation
+		$auto_updates_disabled = defined( 'AUTOMATIC_UPDATER_DISABLED' ) && AUTOMATIC_UPDATER_DISABLED;
+		// The option returns false when nothing was dismissed
 		self::$jetpack_hide_jitm = Jetpack_Options::get_option( 'hide_jitm' );
-		$showphoton = empty( self::$jetpack_hide_jitm['photon'] ) ? 'show' : self::$jetpack_hide_jitm['photon'];
-		$showmanage = empty( self::$jetpack_hide_jitm['manage'] ) ? 'show' : self::$jetpack_hide_jitm['manage'];
-		$show_manage_pi = empty( self::$jetpack_hide_jitm['manage-pi'] ) ? 'show' : self::$jetpack_hide_jitm['manage-pi'];
-		$show_editor = empty( self::$jetpack_hide_jitm['editor'] ) ? 'show' : self::$jetpack_hide_jitm['editor'];
-		if ( 'media-new.php' == $pagenow && ! Jetpack::is_module_active( 'photon' ) && 'hide' != $showphoton ) {
-			add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
-			add_action( 'post-plupload-upload-ui', array( $this, 'photon_msg' ) );
+		// so if it's not an array, it means no JITM was dismissed
+		if ( ! is_array( self::$jetpack_hide_jitm ) ) {
+			if ( 'media-new.php' == $pagenow && ! Jetpack::is_module_active( 'photon' ) ) {
+				add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
+				add_action( 'post-plupload-upload-ui', array( $this, 'photon_msg' ) );
+			}
+			else if ( 'update-core.php' == $pagenow && ! $auto_updates_disabled && ! Jetpack::is_module_active( 'manage' ) ) {
```

Comment:
Instead of adding the `$auto_updates_disabled` checks to every conditional, we should instead return early in the method if it's disabled.